source/Scenes/Backgrounds/SpaceStars.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Annex.Graphics;
+using Annex.Graphics.Contexts;
+using System;
+using System.Collections.Generic;
+
+namespace Game.Scenes.Backgrounds
+{
+    public class SpaceStars : IDrawableObject
+    {
+        private readonly BatchTextureContext _batchTextureContext;
+        private const int MaxStarSize = 16;
+        private const int SpaceSize = 1000;
+
+        public SpaceStars()
+        {
+            var random = new Random();
+
+            var positions = new List<(float, float)>();
+            var rect = new List<(int, int, int, int)>();
+            var sizes = new List<(float, float)>();
+            for (var y = 0; y < SpaceSize; y++)
+            {
+                for (var x = 0; x < SpaceSize; x++)
+                {
+                    var renderStar = random.NextDouble() >= 0.8;
+                    if (!renderStar) continue;
+
+                    var position = random.Next(1, SpaceSize);
+                    positions.Add((x * position, y * position));
```

Comment:
I think it might be better if the SpaceStars could take in some parameters for initialization. Even though the BatchTextureContext is already a huge performance gain compared to TextureContext, we ideally don't want to be generating any more stars than we need to. Here you seem to be placing stars anywhere from [0,0] to [1,000,000,1,000,000]. It'd be great if these bounds were configurable, and if we could specify how many stars we want in those bounds.

Reply:
Definitely a better approach, thanks for explaining :)